### DISCUSSION TITLE
Branch Filter for Environment Protection Rules

### DISCUSSION THREAD

**Discussion by shawnsparks-work** [community/community#27117] [Actions]
<p>The recently announced environment protection rules and environment secrets feature is interesting. However, one big capability it is missing is the ability to specify a branch filter as an environment protection rule. Example: Specify workflows triggered from main branch are the only ones to access certain environment secrets. This way it can be combined with existing branch protection rules to only deploy code that’s been reviewed and merged to a specific branch like main.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/2020-12-15-github-actions-environments-environment-protection-rules-and-environment-secrets-beta/" target="_blank" rel="noopener nofollow ugc">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/443;"><img src="https://user-images.githubusercontent.com/46322400/181101766-1f800c09-521c-4ea0-9718-ce317c8350f8.png" class="thumbnail" width="690" height="443" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/d/9d934875b8b01850d442f5f19e34d9a56905bb06_2_690x443.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/d/9d934875b8b01850d442f5f19e34d9a56905bb06_2_1035x664.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/d/9d934875b8b01850d442f5f19e34d9a56905bb06.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/d/9d934875b8b01850d442f5f19e34d9a56905bb06_2_10x10.png"></div>

<h3><a href="https://github.blog/changelog/2020-12-15-github-actions-environments-environment-protection-rules-and-environment-secrets-beta/" target="_blank" rel="noopener nofollow ugc">GitHub Actions: Environments, environment protection rules and environment...</a></h3>

<p>GitHub Actions: Environments, environment protection rules and environment secrets (beta)</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by chrispat** [comment]
<p>Yes this is one of the items we want to add before we exist the beta.</p>

**Comment by theoribeiro** [comment]
<p><a class="mention" href="/chrispat">@chrispat</a> Are there any plans of rolling this out to private repos of paid plans other than Enterprise?</p>

**Comment by brianmalachiarts** [comment]
<p><a class="mention" href="/chrispat">@chrispat</a> I have the same question: are Environments (including protection rules, required reviewers and env secrets) going to be available for private repos in other paid plans besides Enterprise level?</p>
<p>Our org is on the <strong>Github Team</strong> plan and we would really like to take advantage of these features. We’re trying to show that we can migrate from Jenkins to GitHub Actions completely.</p>
<p>Thanks!</p>

**Comment by dopplershift** [comment]
<p>Is there any way to make branch filter rules apply to tags?</p>

**Comment by chrispat** [comment]
<p>Given GitHub does not have a concept of protected tags we decided to only include branches in the matching for Environments.  If protected tags come in the figure we will likely update the environment rules.</p>

**Comment by dopplershift** [comment]
<p>While I get that, given that builds can be kicked off by a push to a tag (and a release), it would be <em>REALLY</em> helpful to be able to apply to tags so that you can control what tags are allowed to kick off certain builds–regardless of a tags “protected” status.</p>

**Comment by shawnsparks-work** [comment, marked as answer]
<p>I wanted to point that GitHub has released this functionality in case anybody missed it or ends up here later.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/2021-02-17-github-actions-limit-which-branches-can-deploy-to-an-environment/" target="_blank" rel="noopener nofollow ugc">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:640/500;"><img src="https://user-images.githubusercontent.com/46322400/181101771-8d924042-0e83-4acd-b19d-aadff81c4d3a.gif" class="thumbnail animated" width="640" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/a/3a11050962568d90eb0766a48788c3cadf23280b_2_10x10.png"></div>

<h3><a href="https://github.blog/changelog/2021-02-17-github-actions-limit-which-branches-can-deploy-to-an-environment/" target="_blank" rel="noopener nofollow ugc">GitHub Actions: Limit which branches can deploy to an environment | GitHub...</a></h3>

<p>GitHub Actions: Limit which branches can deploy to an environment</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
